packages/elements/src/pagination/index.ts

```diff
@@ -607,27 +446,32 @@ export class Pagination extends BasicElement {
   }
 
   /**
-   * Handles key press event
-   * @param event Key down event object
+   * Handles action when input focused change
+   * @param event focus change event
    * @returns {void}
    */
-  private onKeyPress (event: KeyboardEvent): void {
-    // Allow input number only
-    if (!(/\d/).test(event.key)) {
-      event.preventDefault();
+  private onFocusedChanged (event: FocusEvent): void {
+    this.inputFocused = event.type === 'focus';
+    if (!this.inputFocused) {
+      this.setPage(this.input.value);
     }
   }
 
   /**
-   * Handles action when input focused change
-   * @param event focus change event
+   * Runs on input element `input` event
+   * @param event `input` event
    * @returns {void}
    */
-  private onFocusedChanged (): void {
-    this.inputFocused = document.activeElement === this;
+  protected onInputInput (): void {
+    const currentInput = this.input.value;
+    const inputValue = this.input.value.replace(/[^\d]/g, ''); // stripe invalid charactors
 
-    if (!this.inputFocused) {
-      this.updatePageInput();
+    // Page value cannot start with `0`, reset it if found.
```

Comment:
Why? leading `0` is a valid digit and it does not obstruct entering pages.

Reply:
The valid value of pagination must start at 1

Reply:
OK, than do not set the value to empty string, but just strip leading 0

Reply:
Yes, the strip leading zero is better, Thanks.